wrappers/javascript/packages/anoncreds-nodejs/src/NodeJSAnoncreds.ts

```diff
@@ -617,6 +627,380 @@ export class NodeJSAnoncreds implements Anoncreds {
+    })
+
+    const attributeRawValues = StringListStruct({
+      count: Object.keys(options.attributeRawValues).length,
+      data: Object.values(options.attributeRawValues) as unknown as TypedArray<string>
+    })
+
+    let revocationConfiguration
+    if (options.revocationConfiguration) {
+      const { revocationRegistryDefinition, revocationRegistryDefinitionPrivate, revocationStatusList, registryIndex } =
+        serializeArguments(options.revocationConfiguration)
+
+      revocationConfiguration = CredRevInfoStruct({
+        reg_def: revocationRegistryDefinition,
+        reg_def_private: revocationRegistryDefinitionPrivate,
+        status_list: revocationStatusList,
+        reg_idx: registryIndex
+      })
+    }
+
+    const credentialPtr = allocatePointer()
+    this.nativeAnoncreds.anoncreds_create_w3c_credential(
+      credentialDefinition,
+      credentialDefinitionPrivate,
+      credentialOffer,
+      credentialRequest,
+      attributeNames as unknown as Buffer,
+      attributeRawValues as unknown as Buffer,
+      encoding,
+      revocationConfiguration?.ref().address() ?? 0,
+      credentialPtr
+    )
+    this.handleError()
+
+    return new ObjectHandle(handleReturnPointer<number>(credentialPtr))
+  }
+
+  public processW3CCredential(options: {
+    credential: ObjectHandle
+    credentialRequestMetadata: ObjectHandle
+    linkSecret: string
+    credentialDefinition: ObjectHandle
+    revocationRegistryDefinition?: ObjectHandle | undefined
+  }): ObjectHandle {
+    const { credential, credentialRequestMetadata, linkSecret, credentialDefinition } = serializeArguments(options)
+
+    const ret = allocatePointer()
+
+    this.nativeAnoncreds.anoncreds_process_w3c_credential(
+      credential,
+      credentialRequestMetadata,
+      linkSecret,
+      credentialDefinition,
+      options.revocationRegistryDefinition?.handle ?? 0,
+      ret
+    )
+    this.handleError()
+
+    return new ObjectHandle(handleReturnPointer<number>(ret))
+  }
+
+  public createW3CPresentation(options: {
+    presentationRequest: ObjectHandle
+    credentials: NativeCredentialEntry[]
+    credentialsProve: NativeCredentialProve[]
+    linkSecret: string
+    schemas: Record<string, ObjectHandle>
+    credentialDefinitions: Record<string, ObjectHandle>
+  }): ObjectHandle {
+    const { presentationRequest, linkSecret } = serializeArguments(options)
+
+    const credentialEntries = options.credentials.map((value) =>
+      CredentialEntryStruct({
+        credential: value.credential.handle,
+        timestamp: value.timestamp ?? -1,
+        rev_state: value.revocationState?.handle ?? 0
+      })
+    )
+
+    const credentialEntryList = CredentialEntryListStruct({
+      count: credentialEntries.length,
+      data: credentialEntries as unknown as TypedArray<
+        StructObject<{
+          credential: number
+          timestamp: number
+          rev_state: number
+        }>
+      >
```

Comment:
I think there's a lot of code here for w3c methods that is basically the same except for the end result. Do you think we can reuse some code here? I think this can lead to inconsistencies in functionality

Reply:
I added helper methods doing conversion of parameters in order to reduce code duplication. 
But left w3c* methods. 
I think methods here should be mapped one to one with external C API. Do not see a sense to combine w3c and legacy methods.